docs/source/index.rst

```diff
@@ -130,6 +130,7 @@ To get started, see
    library-user-guide/extending-operators
    library-user-guide/profiling
    library-user-guide/query-optimizer
+   library-user-guide/api-health
```

Comment:
I double checked this is the location in  https://github.com/apache/datafusion/pull/13083